common/src/main/java/org/apache/nemo/common/ir/vertex/LoopVertex.java

```diff
@@ -34,7 +34,7 @@
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
-import java.util.function.IntPredicate;
+import java.util.function.IntPredicate; //NOSONAR
```

Comment:
Let's revert this change on the import line. This isn't applicable. Instead, let's make the `terminationCondition` variable transient on line 59. You can declare it as `private transient IntPredicate`... there and it will solve the issue 👍